_posts/2018-03-16-first-steps-into-gen-servers.md

```diff
@@ -0,0 +1,77 @@
+---
+layout: post
+title:  "Automating Processes with Gen Servers: A Beginner's Approach"
+date:   2018-03-16 10:00:00 -0400
+categories: elixir automation microservices
+author: Kelvin Ma
+---
+As a financial services company, Forward Financing invests a signficant amount of time and resources in timebound reporting -- think daily, weekly, monthly, etc., report generation. As opportunities for automation are concerned, it doesn't get more obvious than this.
+
+Getting out of my comfort zone has been a key part of my engineering growth, using new and unfamiliar tools to build potentially useful tools. After doing a little research on potential solutions for our automation opportunity, I settled on Elixir's GenServer, or Generic Server, pattern. Coming from a primarily Object Oriented Programming background with Rails, this pattern was a bit foreign to me, to say the least.
+
+### Getting Started
+
+The basic premise for this project was to spin up a simple Phoenix/Elixir microservice application to perform two main tasks -- make an API call to one of our monolith apps for portfolio data and then generate an email with the results once a month. After building out the API call in Rails, it was time to get started on the GenServer.
+
+From the [GenServer](https://hexdocs.pm/elixir/GenServer.html) HexDocs:
+
+> A GenServer is a process like any other Elixir process and it can be used to keep state, execute code asynchronously and so on.
+
+Seems pretty straightforward. By writing a GenServer and isolating this single task into its own process, I am effectively creating a microservice that can be managed by the application's own [supervision tree](https://hexdocs.pm/elixir/Supervisor.html).
+
+Here's what I ended up with, using the default callbacks for GenServer:
+
+```
```

Comment:
@kelvinma Can we use syntax highlighting like - https://jekyllrb.com/docs/posts/#highlighting-code-snippets
We might have elixir syntax highlighting and it is inbuilt in jeykll.

Reply:
WOOT! Pretty!